terraso_backend/audit_logs/models.py

```diff
@@ -0,0 +1,49 @@
+from django.db import models
+
+
+# Create your models here.
+class Log(models.Model):
+    """
+    Log model for audits logs
+    """
+    timestamp = models.DateTimeField(auto_now_add=True)
+    keyValueCache = models.JSONField()
+    client_timestamp = models.DateTimeField()
+
+    def __str__(self):
+        return str(self.client_timestamp) + " - " + self.keyValueCache
+
+    def get_string(self, template: str = None) -> str:
+        if template is None:
+            return str(self)
+        return template.format(**self.keyValueCache)
+
+
+# TODO - Discuss: if we should just store string for the the log resource or
+# foreing keys if we store foreing key they will have to be generic keys
+class KeyValue(models.Model):
+    """
+    Key Value model for audits logs
+    """
+    key = models.CharField(max_length=100)
+    value = models.CharField(max_length=500)
+    log = models.ForeignKey(Log, on_delete=models.CASCADE)
+
+    def __str__(self):
+        return self.key + ' - ' + self.value
+
+
+class AuditLog(models.Model):
+    """
+    Audit Log model for audits
+    Note: This model is not used in the current implementation but is the alternative
+    to the KeyValue model and has the same asumption as the KeyValue model that stores
+    the references as strings and not as foreing keys
+    """
+    user = models.CharField(max_length=100)
+    action = models.CharField(max_length=100)
+    resource = models.CharField(max_length=500)
+    timestamp = models.DateTimeField(auto_now_add=True)
```

Comment:
This is what I mentioned in our conversation on the call. To expand on this specific model is not being used just to show how the alternative model could look (for comparison purposes). However, under the assumption that we like this version better, idk if hardcoding tables of logs per resource is the best course of action as the project grows or even today for cases like finding out what action a specific user took will require merging data from two different tables. Secondly, currently, I'm just storing strings that are meant to be uniquely expanded in the comment [above](https://github.com/techmatters/terraso-backend/pull/404#discussion_r1163391530) with the same alternative as above of utilizing GenericForeingKeys

Reply:
can you expand what you mean by "for cases like finding out what action a specific user took will require merging data from two different tables"? is it that there are specific joins you see causing performance issues?

i hear the concern about not wanting the number of audit log tables to keep growing. as far as i know, even in the far future LandPKS modules that we aren't implementing there is never any entity besides project and site to keep logs of, & i don't think there's any intention to add audit logs to any other Terraso features. does it still seem like a concern if we know it will never grow beyond 2 tables?